packages/gestalt/src/ModalAlert.js

```diff
@@ -0,0 +1,244 @@
+      | SyntheticKeyboardEvent<HTMLAnchorElement>
+      | SyntheticKeyboardEvent<HTMLButtonElement>,
+      {| dangerouslyDisableOnNavigation: () => void |},
+    >,
+    rel?: 'none' | 'nofollow',
+    target?: null | 'self' | 'blank',
+  |},
+|};
+
+const ICON_COLOR_MAP = {
+  error: {
+    icon: 'workflow-status-problem',
+    color: 'error',
+  },
+  warning: {
+    icon: 'workflow-status-warning',
+    color: 'warning',
+  },
+};
+
+function Header({
+  accessibilityDismissButtonLabel,
+  type,
+  heading,
+  onDismiss,
+}: {|
+  accessibilityDismissButtonLabel: string,
+  type: 'default' | 'warning' | 'error',
+  heading: string,
+  onDismiss: () => void,
+|}) {
+  return (
+    <Box padding={6}>
+      <Flex flex="grow" alignItems="center" gap={4}>
+        {type !== 'default' && (
+          <Box>
+            <Icon
+              size="20"
+              accessibilityLabel={type}
+              icon={ICON_COLOR_MAP[type].icon}
+              color={ICON_COLOR_MAP[type].color}
+            />
+          </Box>
+        )}
+        <Flex.Item flex="grow">
+          <Heading size="400" accessibilityLevel={1}>
+            {heading}
+          </Heading>
+        </Flex.Item>
+        {type === 'default' && (
+          <Box marginStart={2}>
+            <IconButton
+              accessibilityLabel={accessibilityDismissButtonLabel}
+              bgColor="white"
+              icon="cancel"
+              iconColor="darkGray"
+              onClick={onDismiss}
+              size="sm"
+            />
+          </Box>
+        )}
+      </Flex>
+    </Box>
+  );
+}
+
+function ModalAlertAction({ data, type }: {| data: ActionDataType, type: string |}): Node {
+  const color = type === 'primary' ? 'red' : 'gray';
+  const { accessibilityLabel, disabled, label, onClick, href, rel, target } = data;
+  return href ? (
+    <Button
+      accessibilityLabel={accessibilityLabel}
+      color={color}
+      disabled={disabled}
+      href={href}
+      fullWidth
+      onClick={onClick}
+      iconEnd="visit"
+      rel={rel}
+      role="link"
+      size="lg"
+      target={target}
+      text={label}
+    />
+  ) : (
+    <Button
+      accessibilityLabel={accessibilityLabel}
+      disabled={disabled}
+      color={color}
+      onClick={onClick}
+      fullWidth
+      role="button"
+      size="lg"
+      text={label}
+    />
+  );
+}
+
+/**
+ * An ModalAlert is a simple modal dialog used to alert a user of an issue, or to request confirmation after a user-triggered action. ModalAlert overlays and blocks page content until it is dismissed by the user.
+ */
+export default function ModalAlert({
+  accessibilityDismissButtonLabel,
+  accessibilityModalLabel,
+  type = 'default',
+  children,
+  onDismiss,
+  heading,
+  primaryAction,
+  secondaryAction,
+}: Props): Node {
+  const { accessibilityDismissButtonLabel: accessibilityDismissButtonLabelDefault } =
+    useDefaultLabelContext('ModalAlert');
+
+  [
```

Comment:
You can simplify this data structure a bit:
```js
for (const [key, value] of Object.entries({ primaryAction, secondaryAction })) {
  if ([value?.href, value?.onClick].every((item) => item === undefined)) {
    throw new Error(
        `Either an \`href\` or an \`onClick\` handler must be provided to \`${key}\`.`,
      );
  }
}
```

Reply:
Ooo good call, thanks!